src/Apps/Order/Dialogs.tsx

```diff
@@ -0,0 +1,157 @@
+import { Link } from "@artsy/palette"
+import { ModalDialog, ModalDialogProps } from "Components/Modal/ModalDialog"
+import React from "react"
+import { Container, Subscribe } from "unstated"
+
+interface DialogState {
+  props: ModalDialogProps
+  forceCloseDialog: () => Promise<void>
+}
+
+export class DialogContainer extends Container<DialogState> {
+  state: DialogState = {
+    props: {
+      show: false,
+      heading: null,
+      detail: null,
+      primaryCta: {
+        text: "",
+        action: () => void 0,
+      },
+    },
+    forceCloseDialog: () => Promise.resolve(),
+  }
+
+  // closes the dialog without resolving consumers' promises
+  private closeDialog() {
+    return new Promise(resolve => {
```

Comment:
I wonder if promises are necessary throughout this file and if the same behavior could be achieved in a synchronous way?

Reply:
`setState` is asynchronous and so is showing a user a dialog box and waiting for them to click somewhere, which are the two things I've used promises to model in this file. We'd probably be safe not waiting for setState to complete in the places it's used here, but better safe than sorry IMO. Other than that we could convert to callbacks which would mean we can test synchronously, but I really prefer promises as an API.